Client/Frontend/Browser/BrowserViewController/BrowserViewController+ProductNotification.swift

```diff
@@ -0,0 +1,186 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import BraveShared
+import BraveUI
+import Shared
+
+// MARK: - ProductNotification
+
+extension BrowserViewController {
+    
+    // MARK: Internal
+    
+    @objc func presentEducationalProductNotifications() {
+        guard let selectedTab = tabManager.selectedTab, !benchmarkNotificationPresented else { return }
+        
+        let todayInSeconds = Date().timeIntervalSince1970
+        let checkDate = Preferences.ProductNotificationBenchmarks.ongoingEducationCheckDate.value
+        let isProductNotificationsValid = todayInSeconds <= checkDate
+        
+        var notificationShown = false
+        let contentBlockerStats = selectedTab.contentBlocker.stats
+
+        // Step 1: First Time Block Notification
+        if isProductNotificationsValid,
+           !Preferences.ProductNotificationBenchmarks.firstTimeBlockingShown.value,
+           contentBlockerStats.total > 0 {
+            
+            notifyFirstTimeBlock(theme: Theme.of(selectedTab))
+            
+            Preferences.ProductNotificationBenchmarks.firstTimeBlockingShown.value = true
+            Preferences.ProductNotificationBenchmarks.ongoingEducationCheckDate.value = Date().timeIntervalSince1970 + 7.days
+            
+            notificationShown = true
+        }
+        
+        // Step 2: Load a video on a streaming site
+        guard !notificationShown else { return }
```

Comment:
I used guard here to indicate that the logic should stop in case there is an active notification shown. But I agree the code should be consistent so I am making the change.